pandas/core/indexes/range.py

```diff
@@ -1041,7 +1041,11 @@ def _arith_method(self, other, op):
                 rstop = op(left.stop, right)
 
             res_name = ops.get_op_result_name(self, other)
-            result = type(self)(rstart, rstop, rstep, name=res_name)
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", DeprecationWarning)
+                # The constructor validation can lead to a DeprecationWarning
+                #  from numpy, e.g. with RangeIndex + np.datetime64("now")
```

Comment:
umm shouldn't we do something about this? (e.g. cast to object)

Reply:
yes, turns out to be pretty easy, will handle before long